ui/handle_input.go

```diff
@@ -248,42 +307,91 @@ func (win *Window) HandleInputs() {
 		}
 	}
 
-	for win.inputs.receiveIcons.copy.Clicked(win.gtx) {
+	if win.inputs.receiveIcons.copy.Clicked(win.gtx) {
 		clipboard.WriteAll(win.walletInfo.Wallets[win.selected].Accounts[win.selectedAccount].CurrentAddress)
 		win.addressCopiedLabel.Text = "Address Copied"
 		time.AfterFunc(time.Second*3, func() {
 			win.addressCopiedLabel.Text = ""
 		})
 		return
 	}
+
+	if strings.Trim(win.inputs.spendingPassword.Text(), " ") != "" {
+		strength := dcrlibwallet.ShannonEntropy(win.inputs.spendingPassword.Text()) / 4.0
+		switch {
+		case strength > 0.6:
+			win.outputs.passwordBar.ProgressColor = win.theme.Color.Success
+		case strength > 0.3 && strength <= 0.6:
+			win.outputs.passwordBar.ProgressColor = color.RGBA{248, 231, 27, 255}
+		default:
+			win.outputs.passwordBar.ProgressColor = win.theme.Color.Danger
+		}
+
+		win.outputs.passwordBar.Progress = strength * 100
+	}
 }
 
 func (win *Window) validatePasswords() string {
-	pass := win.inputs.spendingPassword.Text()
+	pass := win.validatePassword()
 	if pass == "" {
-		win.outputs.spendingPassword.HintColor = win.theme.Color.Danger
-		return pass
+		return ""
 	}
 
 	match := win.inputs.matchSpending.Text()
-	if match == "" {
+	if strings.Trim(match, " ") == "" {
 		win.outputs.matchSpending.HintColor = win.theme.Color.Danger
+		win.err = "Enter new wallet password again and it cannot be empty."
 		return ""
 	}
 
 	if match != pass {
-		win.outputs.matchSpending.Color = win.theme.Color.Danger
+		win.err = "New wallet passwords does no match. Try again."
 		return ""
 	}
 
 	return pass
 }
 
+func (win *Window) valOldAndNewPasswords() string {
```

Comment:
I don't see why this is necessary. Password requirements are for dcrlibwallet to handle.